operator/api/v1alpha1/condition_builder.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ConditionBuilder struct {
+	Status metav1.ConditionStatus
+	Reason KymaConditionReason
+}
+
+func NewConditionBuilder() *ConditionBuilder {
+	return &ConditionBuilder{}
+}
+
+func (cb *ConditionBuilder) SetStatus(status metav1.ConditionStatus) {
+	cb.Status = status
+}
+
+func (cb *ConditionBuilder) SetReason(reason KymaConditionReason) {
+	cb.Reason = reason
+}
+
+func (cb *ConditionBuilder) Build() *metav1.Condition {
+	return &metav1.Condition{
+		Type:               string(ConditionTypeReady),
+		Status:             cb.Status,
+		LastTransitionTime: metav1.Time{Time: time.Now()},
+		Reason:             string(cb.Reason),
+		Message:            cb.generateMessage(),
+	}
+}
+
+func (cb *ConditionBuilder) generateMessage() string {
```

Comment:
why is the condition builder code in the api package? I would have assumed this should go into a dedicated `machinery` module or into the kyma controller `pkg` folder. We also have some other candidates for this I think

Reply:
This builder is only used by kyma_types, ideally should not expose to controller, that's why put it in `api` package.
Even controller will use it, I don't think it's controller's responsible to own the implementation, it's part of the contract that api defines and providing to controller.
I assume other candidates you mean those functions with `Kyma` as receiver exists in `kyma_types`, then yes, I also think there should extract to other place, and normally those types definition files only have struct.

Reply:
Lets continue with this now then and after this, do another refactoring for our own machinery package :)